File: server/handler/vo/router.go
package vo

import "github.com/moumou/server/biz/model"

func ConvRouterList2RouterTreeData(routerList []*model.Router) *RouterTreeResponseData {
	data := &RouterTreeResponseData{
		Routers: convRouterListByPid(routerList, 0),
	}
	return data
}

func convRouterListByPid(routerList []*model.Router, pid int) []*RouterRecord {
	routerRecordList := make([]*RouterRecord, 0, len(routerList))
	for _, routerModel := range routerList {
		if int(routerModel.Pid) != pid {
			continue
		}

		routerRecord := convRouter(routerModel)
		routerRecord.Children = convRouterListByPid(routerList, int(routerModel.ID))
		routerRecordList = append(routerRecordList, routerRecord)
	}
	return routerRecordList
}

func convRouter(routerModel *model.Router) *RouterRecord {
	return &RouterRecord{
		Name:   routerModel.Name,
		Title:  routerModel.Title,
		Path:   routerModel.Path,
		IsMenu: routerModel.IsMenu,
	}
}

File: view/src/api/router.ts
import axios from "axios";
import {http} from './request'

export type RouterRecord = {
    name: string
    path: string
    title: string
    is_menu: boolean
    children: RouterRecord[]
}

type RouterTreeResponseData = {
    routers: RouterRecord[]
}

type RouterTreeResponse = {
    code: string
    message: string
    data: RouterTreeResponseData
}

export const routerTreeAPI = async ():Promise<RouterTreeResponse> => {
    let url = '/router_tree'
    const response = await http.post(url)
    return response.data
}